### PR TITLE
Improve metric naming

### PR DESCRIPTION
**Pull request by bboreham**
Addresses some of #2557

### CODE REVIEW

File: prog/weaver/http.go [hunk 1]
```diff
@@ -19,6 +19,8 @@ import (
 	weave "github.com/weaveworks/weave/router"
 )
 
+var allConnectionStates = []string{"established", "pending", "retrying", "failed", "connecting"}
```
Comment (brb) [inline prog/weaver/http.go]: Nitpicking, but I'd prefer `mesh` to provide the list of all possible connection states. Also, this list does not include the `waiting` state (although the if-branch which creates it is marked with the `"shouldn't happen"` comment).

Comment (brb) [inline prog/weaver/http.go]: Probably ignore this, as I've just spotted that the list existed before this PR.

File: prog/weaver/metrics.go [hunk 2]
```diff
@@ -63,15 +61,19 @@ var metrics []metric = []metric{
 	{desc("weave_ips", "Number of IP addresses.", "state"),
 		func(s WeaveStatus, desc *prometheus.Desc, ch chan<- prometheus.Metric) {
 			if s.IPAM != nil {
-				ch <- intGauge(desc, s.IPAM.RangeNumIPs, "total")
 				ch <- intGauge(desc, s.IPAM.ActiveIPs, "local-used")
 			}
 		}},
-	{desc("weave_dns_entries", "Number of DNS entries.", "state"),
+	{desc("weave_max_ips", "Number of IP addresses used by allocator."),
```
Comment (brb) [inline prog/weaver/metrics.go]: I read this description as "# of allocated IPs". Maybe "the size of IP address space" is more accurate?

Comment (bboreham) [inline prog/weaver/metrics.go]: changed

File: prog/weaver/http.go (hunk 1 above)
Comment (bboreham) [inline prog/weaver/http.go]: I agree with your thought; maybe we can do that as part of a more general re-org.

**Comment by brb** [comment]
LGTM.

Re the comment about states, I'm going to file a separate issue for keeping a list of possible improvements.

https://github.com/weaveworks/service/pull/948 needs to be updated.

**Pull request merged**